clis/tcli/src/bin/dtcli/coms.rs

```diff
@@ -0,0 +1,152 @@
+//! Code for handling inter process communication and abstracting away it's dependencies.
+
+use std::{
+    fs,
+    io::{Read, Write},
+    path::Path,
+};
+
+use anyhow::Result;
+use futures::{io::BufReader, AsyncBufReadExt, AsyncWriteExt};
+use interprocess::local_socket::tokio::LocalSocketListener;
+// Purposefully using non async version of this because want to avoid starting async runtime on API.
+use interprocess::local_socket::LocalSocketStream as BlockingSocketStream;
+
+use telio_task::io::{wait_for_tx, Chan};
+
+/// Handler of the DTCLI daemon's side of the IPC communication with the API.
+pub struct DaemonIPC {
+    /// Channel used by the listener task to communicate (send commands received from the IPC socket
+    /// and receive responses to be sent to the IPC socket) with the event_loop.
+    listener_chan: Chan<IPCMessage>,
+}
+
+/// Enum used for controlling the listener task and sending commands and responses as it's payload
+pub enum IPCMessage {
+    /// Variant for wrapping commands to be sent to the event loop.
+    Command(String),
+    /// Variant for wrapping responses to be sent to the IPC listener.
+    Response(String),
+    /// Special variant that sends one last response and stops the IPC listener.
+    QuitResponse(String),
+}
+
+impl DaemonIPC {
+    /// Helper function for creating new instances of Self from mspc halves.
+    ///
+    /// # Arguments
+    ///
+    /// * `listener_chan` - Channel which the event loop uses to communicate.
+    ///
+    /// # Returns
+    ///
+    /// An instance of Self.
+    pub fn new(listener_chan: Chan<IPCMessage>) -> Self {
+        Self { listener_chan }
+    }
+
+    /// Listens for commands (requests) from the API over an IPC socket and
+    /// forwards them to the event loop.
+    /// When the event loop produces a response, sends it back to the API
+    /// over the same IPC socket.
+    /// Note that since a single socket is used, care should be taken to not
+    /// deadlock it.
+    ///
+    /// # Arguments
+    ///
+    /// * `addr` - Address (more specifically path, but left address for clarity) of the IPC socket.
+    pub async fn listen(mut self, addr: &Path) -> Result<()> {
```

Comment:
nit: This could also be renamed to `IPC_sock_fd`

Reply:
Done